darts/ad/anomaly_model.py

```diff
@@ -0,0 +1,1003 @@
+            # TODO: check if model can accept multivariate timeseries, raise error if given and model cannot
+            if "Sequence[darts.timeseries.TimeSeries]" in str(fit_signature_series):
+                self.model.fit(series=list_series, **model_fit_params)
+            else:
+                raise_if_not(
+                    len(list_series) == 1,
+                    f"Forecasting model {self.model.__class__.__name__} only accepts a time series for the training \
+                    phase and not a list of time series.",
+                )
+                self.model.fit(series=list_series[0], **model_fit_params)
+        else:
+            raise_if_not(
+                self.model._fit_called,
+                f"Model {self.model.__class__.__name__} needs to be trained, otherwise set parameter 'allow_model_training' to True \
+                (default: False). The model will then be trained on the given input series.",
+            )
+
+        # generate the historical_forecast() prediction of the model on the train set
+        if self.scorers_are_trainable:
+
+            # check if the window size of the scorers are lower than the max size allowed
+            self._check_window_size(list_series, start)
+
+            list_pred = []
+            for idx, series in enumerate(list_series):
+
+                if list_past_covariates is not None:
+                    past_covariates = list_past_covariates[idx]
+
+                if list_future_covariates is not None:
+                    future_covariates = list_future_covariates[idx]
+
+                list_pred.append(
+                    self._predict_with_forecasting(
+                        series,
+                        past_covariates=past_covariates,
+                        future_covariates=future_covariates,
+                        forecast_horizon=forecast_horizon,
+                        start=start,
+                        num_samples=num_samples,
+                    )
+                )
+
+        # fit the scorers
+        for scorer in self.scorers:
+            if hasattr(scorer, "fit"):
+                scorer.fit(list_pred, list_series)
+
+    def _check_covariates(
+        self,
+        covariates: Union[TimeSeries, Sequence[TimeSeries]],
+        series: Sequence[TimeSeries],
+        name: str,
+    ) -> Sequence[TimeSeries]:
+        """Converts 'covariates' into Sequence, if not already, and checks if its length is equal to the one of 'series'.
+
+        Parameters
+        ----------
+        covariates
+            Covariate ("future" or "past") of 'series'.
+        series
+            The series to be trained on.
+        name
+            Internal parameter for error message, a string indicating if it is a "future" or "past" covariates.
+
+        Returns
+        -------
+        Sequence[TimeSeries]
+            Covariate time series
+        """
+
+        if covariates is not None:
+            covariates = (
```

Comment:
I checked initially, but there is a slight difference: 

series2seq -> return [ts] if isinstance(ts, TimeSeries) else ts
I need ->         return [ts] if not isinstance(ts, Sequence) else ts

It's important because I want the user to be able to input time-series and sequence of time-series.  
And the first thing I do is convert the input into a sequence, and then I iterate through its element to check if they are of type time-series. With series2seq, the user could input something other than a time-series or sequence, and then my iteration will not work, because I may not have a sequence.

Reply:
I see what you mean. In this case I'm wondering whether we should maybe change `series2seq` itself, because its intended use is (AFAIK) exactly this (we use it throughout the code base in places where we want to seamlessly deal with either single TimeSeries or Sequences of TimeSeries).